### PR TITLE
Adds option to specify a deprecation lifetime for builders that support it

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

File: spel/minimal-linux.pkr.hcl [hunk 1]
```diff
@@ -734,6 +741,9 @@ locals {
 
   # Template the description string
   description = "STIG-partitioned [*NOT HARDENED*], LVM-enabled, \"minimal\" %s, with updates through ${formatdate("YYYY-MM-DD", timestamp())}. Default username `maintuser`. See ${var.spel_description_url}."
+
+  # Calculate AWS AMI deprecate_at timestamp
+  aws_ami_deprecate_at = var.spel_duration != null ? timestamp() + var.spel_duration : null
```
Comment (eemperor) [inline spel/minimal-linux.pkr.hcl:746]: Mismatched variable name.
```suggestion
  aws_ami_deprecate_at = var.spel_duration != null ? timestamp() + var.spel_deprecation_lifetime : null
```

Comment (lorengordon) [reply spel/minimal-linux.pkr.hcl]: oops yes, done

**Comment by lorengordon** [comment]
go codebuild go

**Comment by lorengordon** [comment]
go codebuild go

**Review by eemperor — APPROVED** [review]
Looks good to me.  Codebuild appears to need permissions to set the deprecation.

**Comment by lorengordon** [comment]
go codebuild go

**Comment by ferricoxide** [comment]
go codebuild go

**Pull request merged**